File: hospital-be-main/src/HospitalLibrary/Core/DTO/ViewAppointmentByPatientDTO.cs
using HospitalLibrary.Core.Enum;
using HospitalLibrary.Core.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HospitalLibrary.Core.DTO
{
    public class ViewAppointmentByPatientDTO
    {
        public Status Status { get; set; }
        public string DoctorName { get; set; }
        public Guid AppointmentId { get; set; }
        public DateRange Date { get; set; }

        public ViewAppointmentByPatientDTO(Appointment app)
        {
            Status = app.Status;
            Date = app.Range;
            AppointmentId = app.Id;
            DoctorName = app.Doctor.GetFullName();
        }
    }
}

File: hospital-fe-main/src/app/modules/hospital/model/ViewAppointmentByPatientDTO.ts
import {DateRange} from "./date-range.model";

export class ViewAppointmentByPatientDTO {
  doctor: string = "";
  date!: DateRange;
  appointmentId : string = "";

  public constructor(Doctor: string = "", Date: DateRange, AppointmentId: string = ""
  ) {
    this.doctor = Doctor;
    this.date = Date;
    this.appointmentId = AppointmentId;

  }
}
